### PR TITLE
Handle entry_points that end with .py

### PR DESCRIPTION
**Pull request by groodt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
There are some entry_points in the wild that end with a `.py` suffix. This is an extreme edge-case, but this clashes with the native Python rules in Bazel that forbids `py_binary` to end with `.py`.

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
An example entry_point in the wild that ends with `.py` https://github.com/jupyterhub/jupyterhub-idle-culler/blob/9dba16033485f3c70352edd3255bb0c5905f16ce/setup.py#L15

### CODE REVIEW

File: python/pip_install/extract_wheels/lib/bazel.py [hunk 1]
```diff
@@ -409,13 +409,18 @@ def extract_wheel(
     directory_path = Path(directory)
     entry_points = []
     for name, entry_point in sorted(whl.entry_points().items()):
-        entry_point_script = f"{WHEEL_ENTRY_POINT_PREFIX}_{name}.py"
-        (directory_path / entry_point_script).write_text(
+        # There is an extreme edge-case with entry_points that end with `.py`
+        # See: https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/test/java/com/google/devtools/build/lib/rules/python/PyBinaryConfiguredTargetTest.java#L174
+        entry_point_without_py = name[:-3] if name.endswith(".py") else name
```
Comment (thundergolfer) [inline python/pip_install/extract_wheels/lib/bazel.py:414]:
```suggestion
        entry_point_without_py = name.removesuffix(".py")
```

https://docs.python.org/3/library/stdtypes.html#str.removesuffix

**Review by thundergolfer — APPROVED** [review]

File: python/pip_install/extract_wheels/lib/bazel.py (hunk 1 above)
Comment (groodt) [reply python/pip_install/extract_wheels/lib/bazel.py:414]: Incredible as it may seem `New in version 3.9.` and I would prefer this to be compatible with older versions of PY3.

Comment (thundergolfer) [reply python/pip_install/extract_wheels/lib/bazel.py:414]: Bah, so it is.
